code/modules/reagents/chemistry/reagents/impure_reagents.dm

```diff
@@ -1,26 +1,195 @@
 //Reagents produced by metabolising/reacting fermichems inoptimally, i.e. inverse_chems or impure_chems
 //Inverse = Splitting
 //Invert = Whole conversion
+//Generally, make their pHes more acidic if acidic, or more basic if basic, so that you push them towards heartburn
 
+//Causes metabolic stress, and that's it.
 /datum/reagent/impure
+	name = "Chemical Isomers"
+	id = "generic_impure"
+	description = "Toxic chemical isomers made from impure reactions. At low volumes will cause light toxin damage, but as the volume increases, it deals larger amounts, damages the liver, then eventually the heart. This is default impure chem for all chems, and changes only if stated."
 	chemical_flags = REAGENT_INVISIBLE | REAGENT_SNEAKYNAME //by default, it will stay hidden on splitting, but take the name of the source on inverting
+	var/metastress = 1
+	var/obj/item/organ/liver/L
+	pH = 3
 
+/datum/reagent/impure/on_mob_add(mob/living/L)
+	var/mob/living/carbon/C = L
+	if(C && (!(L)))
+		L = C.getorganslot(ORGAN_SLOT_LIVER) //reduce calls
+	..()
+
+/datum/reagent/impure/on_mob_life(mob/living/carbon/C)
+	if(!L)//Though, lets be safe
+		L = C.getorganslot(ORGAN_SLOT_LIVER)
+		C.adjustToxLoss(5, TRUE)//Incase of no liver!
+		return
+	L.adjustMetabolicStress(metastress)
+	..()
 
 /datum/reagent/impure/fermiTox
-	name = "Chemical Isomers"
+	name = "Trans Chemical Isomers"
 	id = "fermiTox"
-	description = "Toxic chemical isomers made from impure reactions. At low volumes will cause light toxin damage, but as the volume increases, it deals larger amounts, damages the liver, then eventually the heart. This is default impure chem for all chems, and changes only if stated."
+	description = "Toxic chemical isomers that will deal direct toxin damage instead of metabolic damage."
 	data = "merge"
 	color = "FFFFFF"
 	can_synth = FALSE
-	var/potency = 1 //potency multiplies the volume when added.
-
+	metastress = 2
+	pH = 2
 
 //I'm concerned this is too weak, but I also don't want deathmixes.
 //TODO: liver damage, 100+ heart
 /datum/reagent/impure/fermiTox/on_mob_life(mob/living/carbon/C, method)
-	if(C.dna && istype(C.dna.species, /datum/species/jelly))
-		C.adjustToxLoss(-2)
-	else
-		C.adjustToxLoss(2)
+	C.adjustToxLoss(metastress, TRUE)
+	..()
+
+/datum/reagent/impure/mannitol
+	name = "Mannitoll"
+	id = "mannitol_impure"
+	description = "Inefficiently causes brain damage."
+	color = "#CDCDFF"
+	pH = 12.4
+
+/datum/reagent/impure/mannitol/on_mob_life(mob/living/carbon/C)
+	C.adjustOrganLoss(ORGAN_SLOT_BRAIN, cached_purity*REM)
+	..()
+
+//I am incapable of making anything simple
+/datum/reagent/impure/neurine
+	name = "Neruwhine"
+	id = "neurine_impure"
+	description = "Causes the patient a temporary trauma."
```

Comment:
Weird wording? Something like "Causes a temporary trauma in the patient" would make more sense but well it's not like the description ever comes up

Reply:
Fixed